modules/text/CMakeLists.txt

```diff
@@ -3,12 +3,46 @@ ocv_define_module(text opencv_ml opencv_imgproc opencv_core opencv_features2d op
 
 if(NOT CMAKE_CROSSCOMPILING OR OPENCV_FIND_TESSERACT)
   set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+  if(COMMAND pkg_check_modules)
+    pkg_check_modules(Tesseract tesseract)
+  endif()
   find_package(Tesseract QUIET)
+
   if(Tesseract_FOUND)
-    message(STATUS "Tesseract:   YES")
+    set(Tesseract_LIBRARY ${Tesseract_LIBRARIES})
+    set(Tesseract_INCLUDE_DIR ${Tesseract_INCLUDE_DIRS})
+  else()
+    find_path(Tesseract_INCLUDE_DIR tesseract/baseapi.h
+      HINTS
+      /usr/include
+      /usr/local/include)
+
+    find_library(Tesseract_LIBRARY NAMES tesseract
+      HINTS
+      /usr/lib
+      /usr/local/lib)
+  endif()
+
+  find_package(Leptonica QUIET)
+  if(Leptonica_FOUND)
+    set(Lept_LIBRARY ${Leptonica_LIBRARIES})
+  else()
+    find_library(Lept_LIBRARY NAMES lept
+      HINTS
+      /usr/lib
+      /usr/local/lib)
+  endif()
+  if(Lept_LIBRARY)
+    message(STATUS "Found Tesseract dependencies: Leptonica ${Leptonica_VERSION}")
+  endif()
+ 
+  if(Tesseract_INCLUDE_DIR AND Tesseract_LIBRARY AND Lept_LIBRARY)
+    message(STATUS "Tesseract:   YES ${Tesseract_VERSION}")
     set(HAVE_TESSERACT 1)
-    ocv_include_directories(${Tesseract_INCLUDE_DIRS})
-    ocv_target_link_libraries(${the_module} ${Tesseract_LIBRARIES})
+ 
+    ocv_include_directories(${Tesseract_INCLUDE_DIR})
+    ocv_target_link_libraries(${the_module} ${Tesseract_LIBRARY})
```

Comment:
> Tesseract_LIBRARIES

Replacing to `Tesseract_LIBRARY` is wrong.
Proper `find_package(Tesseract)` script brings into `LIBRARIES` variable all dependencies required for **static** build.
"Proper" means created correctly.

OpenCV doesn't use leptonica directly, so it should not care about it. The same note is about mentioned libpng, zlib, tiff (no conflicts?) libraries too.


Example (Fedora 29):
```
$ pkgconf --libs tesseract
-ltesseract
$ pkgconf --static --libs tesseract
-ltesseract -lpthread -llept -lz -lpng16 -lz -ljpeg -lgif -ltiff -lwebp
```

All transitive dependencies are handled via `pkgconfig`.